app/src/main/java/com/mdg/droiders/samagra/shush/AlarmScheduler.java

```diff
@@ -30,52 +36,299 @@ public AlarmScheduler(Context context) {
     }
 
     /**
+     * Sets an alarm to shush the phone for a single day.
+     *
+     * @param startTime The time at which phone is to be shushed
+     * @param endTime   The time at which phone is to be un-shushed
+     * @param day       int type for week days starting from monday and zero indexed.
+     *                  For e.g. - Monday is 0, Tuesday is 1, ... and so on
+     * @param rowID     Unique primary key of the shush alarm row.
+     */
+    public void setSingleDayAlarm(Calendar startTime, Calendar endTime, int day, Integer rowID) {
+        if (day > 6 || rowID == null) {
+            return;
+        }
+        boolean[] days = new boolean[7];
+        for (int i = 0; i < 7; i++) {
+            days[i] = day == i;
+        }
+        setWeeklyAlarm(startTime, endTime, days, rowID);
+    }
+
+    /**
+     * Cancels a shush alarm for a particular day if it exists.
+     *
+     * @param day   int type for week days starting from monday and zero indexed.
+     *              For e.g. - Monday is 0, Tuesday is 1, ... and so on
+     * @param rowID Unique primary key of the shush alarm row.
+     */
+    public void cancelSingleAlarm(int day, Integer rowID) {
+        if (rowID == null) {
+            return;
+        }
+        alarmManager.cancel(getDefaultPendingIntent(getStartDayID(day, rowID)));
+        alarmManager.cancel(getDefaultPendingIntent(getEndDayID(day, rowID)));
+    }
+
+    /**
+     * Sets a weekly alarm to silence your phone for days given by boolean array days.
+     * <br>
+     * <b>NOTE</b> : This method only schedules alarms and does NOT cancels for some day whose
+     * corresponding index in days array is false. To cancel alarms
+     * use {@link #cancelSingleAlarm(int, Integer)}
+     * <br><br>
+     * <b>Uses</b> : {@link #setAlarm(long, long, Integer, Integer)} to set each alarm one by one.
+     *
+     * @param startTime The time of day at which phone is to be shushed
+     * @param endTime   The time of day at which phone is to be un-shushed
+     * @param days      A boolean array that is true if the alarm is to be
+     *                  scheduled for the corresponding day. i.e. days[0] is set to be true
+     *                  then a weekly alarm will be set such that it will shush the phone every Monday.
+     * @param rowID     Unique primary key of the shush alarm row.
+     */
+    public void setWeeklyAlarm(Calendar startTime, Calendar endTime, boolean[] days, Integer rowID) {
+        // To avoid various exceptions
```

Comment:
Why isn't an exception thrown or logged here?

Reply:
I will log in the error stream in next commit.